### PR TITLE
Project-mongo-api by Linnéa Helén

### PR DESCRIPTION
**Pull request by Polkapetunian**
Root endpoint

https://umea-trees.herokuapp.com/

This is a project with trees cared for by Umeå park department.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,13 +1,32 @@
-# Mongo API Project
+# City of birches
```
Comment (TheErry) [inline README.md:1]: Nice name! :D

File: README.md [hunk 2]
```diff
@@ -1,13 +1,32 @@
...
 
-Replace this readme with your own information about your project. 
-
-Start by briefly describing the assignment in a sentence or two. Keep it short and to the point.
+The goals of this project were to practise modelling data in Mongoose, fetching items from a Mongo database using Mongoose and seeding a large amount of data to a database.
 
 ## The problem
 
-Describe how you approached to problem, and what tools and techniques you used to solve it. How did you plan? What technologies did you use? If you had more time, what would be next?
+I wanted to use some data of interest to me. As I have studied landscape architecture I am familiar with trees. I found open data from municipality of Umeå with trees that are taken care of by the Park department. The original json that I could download from their open data site was a bit complicated with weird param naming and lots of nested objects and arrays. So I downloaded an excel file instead and changed it up a bit. Then I saved it as a CSV file and used a web app, csvjson.com to convert it to json.
```
Comment (TheErry) [inline README.md:7]: That's awesome that you managed to change the file and make it from CSV to json. Good job! :D

File: README.md [hunk 3]
```diff
@@ -1,13 +1,32 @@
...
+
+https://umea-trees.herokuapp.com/trees
+
+Tree by ID endpoint
+
+https://umea-trees.herokuapp.com/trees/id/:id
+
+All tree species
+
+https://umea-trees.herokuapp.com/trees/species
```
Comment (TheErry) [inline README.md:32]: Great that you have the endpoints listed!

File: server.js [hunk 4]
```diff
@@ -1,39 +1,89 @@
...
+  latitude: mongoose.Decimal128,
+  scientificName: String,
+  treeType: String,
+  streetOrPark: String
+})
+
+const Tree = mongoose.model('Tree', treeSchema)
+
+if (process.env.RESET_DB) {
+  console.log('Resetting database!')
```
Comment (TheErry) [inline server.js:27]: No console.logs in the finished code ;)

File: server.js [hunk 5]
```diff
@@ -1,39 +1,89 @@
...
+      }
+    })
+    res.json(trees)
+  } else {
+    const trees = await Tree.find() 
+    res.json(trees)
+  }})
+
+//TREE BY ID
+app.get('/trees/id/:id', async (req, res) => {
```
Comment (TheErry) [inline server.js:67]: I don't think you need the extra id here for the id-param.

File: server.js [hunk 6]
```diff
@@ -1,39 +1,89 @@
...
+    const trees = await Tree.find({ 
+      scientificName: {
+        $regex: new RegExp(name, "i")
+      }
+    })
+    res.json(trees)
+  } else {
+    const trees = await Tree.find() 
+    res.json(trees)
+  }})
```
Comment (TheErry) [inline server.js:64]: Really nice!

**Review by TheErry — APPROVED** [review]
Great job on the project! I really like that you used data that you found on your own. I wonder who had the energy to count and add all the data for all the trees :D 
I don't have much more to say than great job!

File: README.md [hunk 7]
```diff
@@ -1,13 +1,32 @@
...
+https://umea-trees.herokuapp.com/
+
+All trees endpoint 
+with possibility to query with ?name=betula  for example
+
+https://umea-trees.herokuapp.com/trees
+
+Tree by ID endpoint
+
+https://umea-trees.herokuapp.com/trees/id/:id
```
Comment (KarolCislowski) [inline README.md:28]: `https://umea-trees.herokuapp.com/trees/:id` when you use id do not specify it in your address 🤓

File: README.md (hunk 1 above)
Comment (KarolCislowski) [inline README.md:1]: My first thought when I saw your data was "F@$#, do day have different species than Betulas??"  You played on my sentiment here 😸 You know that I was study Gardening for 1 year in Poland :D  Dendrology was my favorite classes 🌳

File: server.js [hunk 8]
```diff
@@ -1,39 +1,87 @@
...
```
Comment (KarolCislowski) [inline server.js:62]: nice filtering

**Review by KarolCislowski — APPROVED** [review]
Great Project 🥇

**Pull request closed without merging**